src/normalize/core.ts

```diff
@@ -225,14 +230,31 @@ export class CoreNormalizer extends SpecMapper<NormalizerParams, FacetedUnitSpec
       encoding: replaceRepeaterInEncoding(encoding, params.repeater)
     });
 
-    return this.mapUnit(
-      {
-        ...spec,
-        ...(mergedProjection ? {projection: mergedProjection} : {}),
-        ...(mergedEncoding ? {encoding: mergedEncoding} : {})
-      },
-      {config}
-    );
+    if (this.isEmpty(encoding) && !this.isEmpty(params.repeater)) {
```

Comment:
Can we love this check to the second argument of the map unit method?

Reply:
ah yes, simplify and updated!